src/main/java/org/jenkinsci/plugins/edithelp/EditHelpRootAction.java

```diff
@@ -0,0 +1,162 @@
+package org.jenkinsci.plugins.edithelp;
+
+import java.io.*;
+
+import hudson.Extension;
+import hudson.cli.PrivateKeyProvider;
+import hudson.model.Action;
+import hudson.model.RootAction;
+import hudson.model.View;
+import jenkins.model.Jenkins;
+import org.apache.tools.ant.taskdefs.Recorder;
+import org.kohsuke.stapler.Stapler;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.lang.Klass;
+import sun.misc.URLClassPath;
+
+import javax.swing.*;
+import java.net.Socket;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static java.util.logging.Level.FINE;
+
+
+@Extension
+public class EditHelpRootAction implements RootAction {
+
+    private static final Logger LOGGER = Logger.getLogger(PrivateKeyProvider.class.getName());
+    //reading the catalog of helpmanager
+    private static ArrayList<File> listWithFileNames = new ArrayList<>();
+    public static void getListFiles(String str) {
+        File f = new File(str);
+        File[] mydr = f.listFiles();
+        if (mydr == null)
+            return;
+        for (File s : mydr) {
+            if (s.isFile()) {
+                listWithFileNames.add(s);
+            } else if (s.isDirectory()) {
+                getListFiles(s.getAbsolutePath());
+            }
+        }
+    }
+
+    //array for cache storage
+    Map<String, String> map = new TreeMap<String, String>();
+
+    //reading files into cache
+    public EditHelpRootAction() throws IOException, FileNotFoundException {
+        Jenkins jn = Jenkins.getInstance();
+        if (jn != null) {
+            File dirfile = jn.getRootDir();         
+            String dirName = null;         
+            if(dirfile != null) {
+                dirName = dirfile.toString()+"/helpmanager";
+            //check if the directory exists
+                File helpManagerFile = new File(dirName);
+
+                if(!helpManagerFile.exists())
+                    if (!helpManagerFile.mkdirs()) {
+                        LOGGER.log(FINE,"unable create directory");
+                    }
+
+                //reading files from the directory
+                getListFiles(dirName);
```

Comment:
and what next? All methods should return value or do some action. Why get action return nothing?